### PR TITLE
- Converted IPrevalueConnector to IDataTypeConfigurationConnector

### PR DESCRIPTION
**Pull request by mclausen**
- Converted IPrevalueConnector to IDatatTypeConfigurationConnector
- Updated contracts according to v8 specifications of the IDatatTypeConfiguration object
- Renamed Prevalue concepts to Configuration

### CODE REVIEW

File: src/Umbraco.Core/Deploy/IDataTypeConfigurationConnector.cs [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
 namespace Umbraco.Core.Deploy
 {
     /// <summary>
-    /// Defines methods that can convert a data type configurations to / from an environment-agnostic string.
+    /// Defines methods that can convert a configuration to / from an environment-agnostic string.
     /// </summary>
-    /// <remarks>Configurations may contain values such as content identifiers, that would be local
+    /// <remarks>Configuration may contain values such as content identifiers, that would be local
     /// to one environment, and need to be converted in order to be deployed.</remarks>
-    [SuppressMessage("ReSharper", "UnusedMember.Global", Justification = "This is actual only used by Deploy, but we dont want third parties to have references on deploy, thats why this interface is part of core.")]
```
Comment (bergmania) [inline src/Umbraco.Core/Deploy/IDataTypeConfigurationConnector.cs]: Please don't remove this. Otherwize the interface appears as not in use.

File: src/Umbraco.Core/Deploy/IDataTypeConfigurationConnector.cs [hunk 2]
```diff
@@ -17,18 +15,18 @@ public interface IDataTypeConfigurationConnector
         IEnumerable<string> PropertyEditorAliases { get; }
 
         /// <summary>
-        /// Gets the environment-agnostic data type configurations corresponding to environment-specific configurations.
+        /// Gets the environment-agnostic preValues corresponding to environment-specific preValues.
```
Comment (bergmania) [inline src/Umbraco.Core/Deploy/IDataTypeConfigurationConnector.cs]: PreValues is not a concept and should also be updated in comments..

**Comment by zpqrtbnk** [comment]
I don't understand this PR. It's re-introducing "pre-value" wording in Core - but preValues don't exist anymore. It's changing details and ... I don't understand why. And then, at the core of if, using dictionaries for the Core configurations... I think there's something wrong. Looking into it. Might propose an alternative.

File: src/Umbraco.Core/Deploy/IDataTypeConfigurationConnector.cs (hunk 2 above)
Comment (mclausen) [reply src/Umbraco.Core/Deploy/IDataTypeConfigurationConnector.cs]: Didn't seem like i got all of them. Should be fixed

File: src/Umbraco.Core/Deploy/IDataTypeConfigurationConnector.cs (hunk 1 above)
Comment (mclausen) [reply src/Umbraco.Core/Deploy/IDataTypeConfigurationConnector.cs]: Fixed

**Comment by mclausen** [comment]
> I don't understand this PR. It's re-introducing "pre-value" wording in Core - but preValues don't exist anymore. It's changing details and ... I don't understand why. And then, at the core of if, using dictionaries for the Core configurations... I think there's something wrong. Looking into it. Might propose an alternative.

I think I didn't manage to replace all the places pre-values where mentioned. Regarding the idea, was that we would serialize/deserialize the configurations like we do in the ValueConnectors. So that when we iterate the property editors we would parse the configuration to the current configation?

**Comment by zpqrtbnk** [comment]
all in all, I'm going to do it directly in Core, and differently - closing

**Pull request closed without merging**